### ISSUE TITLE
Accessing Json value within RedisValue without deserializing it

### ISSUE THREAD

**Issue by Ekotenggara** [StackExchange/StackExchange.Redis#2092]
Hi, 

Sorry for asking a simple question, I am still putting my head around redis.
I have been using redisjson to store my data and using redisearch to query my data in C# using nRediSearch and StackExchange.Redis.

My query goes like this :

            var mux = ConnectionMultiplexer.Connect("localhost");
            var db = mux.GetDatabase();
            var client = new NRediSearch.Client("ResultSearch", db);

            var tempdocuments = client.Search(new Query("@designId:123")).Documents;

            foreach (var test1 in tempdocuments)
            {
                RedisValue tempval = test1["json"];
                //tempval return something like {{"a":"40450","b":"4070700","c":"4070700","d":"222","e":"7","f":"16048047","g":"10","h":"2022-04-11 16:01:17","i":"A4","j":null,"k":null,"l":"0","m":"1","n":154795.25222315098}}

                //I want to get the n component of tempval 154795.25222315098 without deserializing json, can I do it?

            }

I want to get the n component of tempval 154795.25222315098 without deserializing json, can I do it?
I have a millions of this tempval deserializing each json will penalize my speed heavily. Is there anyway to achieve this?

Thank you,

Eko

**Comment by NickCraver** [comment]
There's nothing special in the lib to be had here, just anything you'd do as a string. For example a regex or other string parsing looking for `"n":`...and the pros and cons that implies. It'll vary on how safe that is to do, but if you want to shortcut there: it's not Redis specific you're just looking to minimally parse JSON :)

Also note that parsing JSON _may_ be a quick-enough option if you do it minimally. For example you don't need to deserialize to `MyFullClass`, you could deserialize to something that's just:
```cs
public class MinimalThing
{
   public double n { get; set; }
}
```
...and see how fast that performs, since the reader is chucking everything else out the window while parsing.

**Comment by slorello89** [comment]
@Ekotenggara, part of the beauty of RedisJSON is that you don't need to do any work deserializing JSON objects :) if you've stored these "Results" in RedisJSON, and have indexed the result you want, you can just use a `RETURN` sub-command within `FT.SEARCH` to return them.

I maintain [Redis OM .NET](https://github.com/redis/redis-om-dotnet) which is a SE.R backed abstraction over in particular RediSearch/JSON, so if your data is somewhat modelable, you should be able to manage this relatively easily:


```csharp
var results = provider.RedisCollection<Result>();
await results.InsertAsync(new Result() {designId = "123", n = 154795.25222315098});
var res = results.Where(x => x.designId == "123").Select(x => x.n).First();
```

You'd need to model your data, and generate your indexed based off that model:

```csharp
[Document(StorageType = StorageType.Json)]
public class Result
{
    [Indexed]
    public string designId { get; set; }
    [Indexed]
    public double n { get; set; }
}
```

With NRediSearch, you can add the Return Fields you are looking for,

```csharp
var query = new Query("@designId:{123}");
query.ReturnFields("$.n");
var tmpDocs = client.Search(query).Documents;

foreach (var doc in tmpDocs)
{
    Console.WriteLine(doc["$.n"]);
}
```

will pull back just `$.n`, which is the path you're looking for.

In other words, there's no real need for any sort of minimal JSON parsing as you get it for free from Redis :)

**Comment by Ekotenggara** [comment]
Thank you everyone, it is all sorted now. Thank you @NickCraver for your nice faster json alternative. Thank you @slorello89 for introducing me to your masterpiece.
